app/questionnaire/answer_store_updater.py

```diff
@@ -13,15 +14,26 @@ def __init__(self, current_location, schema, questionnaire_store, current_questi
         self._schema = schema
         self._questionnaire_store = questionnaire_store
         self._answer_store = self._questionnaire_store.answer_store
+        self._list_store = self._questionnaire_store.list_store
 
-    def save_answers(self, form):
+    def save_answers(self, form, save_completed_blocks=True):
         self._update_questionnaire_store_with_form_data(form.data)
 
-        if self._current_location not in self._questionnaire_store.completed_blocks:
-            self._questionnaire_store.completed_blocks.append(self._current_location)
+        if save_completed_blocks:
+            if self._current_location not in self._questionnaire_store.completed_blocks:
+                self._questionnaire_store.completed_blocks.append(self._current_location)
 
         self._questionnaire_store.add_or_update()
 
+    def add_new_list_item_answers(self, form, list_name):
```

Comment:
To be consistent with the `save_answers` method this should be named `save_new_list_item_answers`.